### PR TITLE
dev/core#1767 Fix phone key parsing in CRM_Dedupe_Finder

### PR DESCRIPTION
**Pull request by jaapjansma**
Overview
----------------------------------------
This is a PR for an issue reported on lab. That issue also contained a solution and I have created the PR. As SRQ_Civicrm was not able to do so. So I helped him by providing the PR. 

The issue is wel documented on lab.

Before
----------------------------------------

The reg ex for finding duplicate phone numbers was wrong.

1. Create individual unsupervised rule that matches on phone
1. Create input profile that creates contact with phone-main-mobile and set to "Update the matching contact" on duplicate match
1. Create multiple contacts with same phone

-> Creates duplicate contacts

After
----------------------------------------

The reg ex has been changed a duplicate phone numbers would be found. 

->Update pre-existing contact

Comments
----------------------------------------
See https://lab.civicrm.org/dev/core/-/issues/1767 for reproduction steps.

### CODE REVIEW

File: CRM/Dedupe/Finder.php [hunk 1]
```diff
@@ -252,7 +252,7 @@ public static function formatParams($fields, $ctype) {
     // the -digit to civicrm_address.location_type_id and -Primary to civicrm_address.is_primary
     foreach ($flat as $key => $value) {
       $matches = [];
-      if (preg_match('/(.*)-(Primary-[\d+])$|(.*)-(\d+|Primary)$/', $key, $matches)) {
+      if (preg_match('/(.\*)-(Primary-[\d+])$|(.\*)-(\d+-\d+)$|(.\*)-(\d+|Primary)$/', $key, $matches)) {
```
Comment (colemanw) [inline CRM/Dedupe/Finder.php:255]: I don't understand the escaping of the star character here. What does that accomplish?

**Review by colemanw — CHANGES_REQUESTED** [review]
Needs updating per comments on gitlab

**Comment by jaapjansma** [comment]
I have updated the PR

**Comment by jaapjansma** [comment]
@colemanw are you able to merge this PR?

**Comment by eileenmcnaughton** [comment]
Ok I managed to put aside some time for this & here is the test https://github.com/civicrm/civicrm-core/pull/17882 - I was able to replicate it in the test & now it has a test it is mergeable

**Comment by colemanw** [comment]
Thanks for doing that @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
@colemanw yeah - @jaapjansma had asked for help on this one so that got it high enough on the list to spend a evening on it

**Pull request merged**